bigint/src/algorithms.rs

```diff
@@ -555,6 +556,124 @@ pub fn ilog2<T: traits::PrimInt>(v: T) -> usize {
     fls(v) - 1
 }
 
+pub struct MontyReducer<'a> {
```

Comment:
Let's put all this in its own `monty` module, instead of adding to `algorithms`.

Also, I realize it wasn't your code to begin with, but if you can figure out some good code comments to add to this implementation, it would help a lot for reviewing and future maintenance.

Reply:
ACK. And yes, I planned to add comments before making the PR, but the strange (self-inflicted) behaviour motivated me to get early feedback :)